queries/tornado_session.py

```diff
@@ -158,6 +158,7 @@ def __init__(self, uri=DEFAULT_URI,
         """
         self._connections = dict()
         self._cleanup_callback = None
+        self._connecting = dict()
```

Comment:
Perhaps this should just be a set of FDs, or perhaps a refactor of self._connections to have a dict of handle and state, where state would toggle between connecting and connected? Maybe that's too much. I'd probably just go with the set. A dict with just True or False is "extra" overhead (albeit minor) that isn't really needed for what you're trying to do.

Reply:
Set makes sense